docs/developer/solutions/archive.rst

```diff
@@ -0,0 +1,152 @@
+Solution archive guidelines
+===========================
+
+To provide a predictable interface with packaged Solutions, MetalK8s expects a
+few criteria to be respected, described below.
+
+
+Archive format
+--------------
+
+Solution archives must use the `ISO-9660:1988`_ format, including `Rock Ridge`_
+and Joliet_ directory records. The character encoding must be UTF-8_. The
+conformance level is expected to be at most 3, meaning:
+
+- Directory identifiers may not exceed 31 characters in length
+- File name + ``'.'`` + file name extension may not exceed 30 characters in
+  length
+- Files are allowed to consist of multiple sections
+
+The generated archive should specify a volume ID, set to
+``{project_name} {version}``.
+
+.. todo::
+
+   Clarify whether Joliet/Rock Ridge records supersede the conformance level
+   w.r.t. filename lengths
+
+
+.. _`ISO-9660:1988`: https://www.iso.org/obp/ui/#iso:std:iso:9660:ed-1:v1:en
+.. _`Rock Ridge`: https://en.wikipedia.org/wiki/Rock_Ridge
+.. _Joliet: https://en.wikipedia.org/wiki/Joliet_(file_system)
+.. _UTF-8: https://tools.ietf.org/html/rfc3629
+
+Here is an example invocation of the common Unix mkisofs_ tool to generate such
+an archive::
+
+   mkisofs
+       -output my_solution.iso
+       -R  # (or "-rock" if available)
+       -J  # (or "-joliet" if available)
+       -joliet-long
+       -l  # (or "-full-iso9660-filenames" if available)
+       -V 'MySolution 1.0.0'  # (or "-volid" if available)
+       -gid 0
+       -uid 0
+       -iso-level 3
+       -input-charset utf-8
+       -output-charset utf-8
+       my_solution_root/
+
+.. _mkisofs: https://linux.die.net/man/8/mkisofs
+
+.. todo::
+
+   Consider if overriding the source files UID/GID to 0 is necessary
+
+
+File hierarchy
+--------------
+
+Here is the file tree expected by MetalK8s to exist in each Solution archive::
+
+   .
+   ├── images
+   │   ├── listing.yaml
+   │   └── some_image_name.tar.gz
+   ├── operator
+   │   ├── crd
+   │   │   └── some_crd_name.yaml
+   │   └── deployment.yaml
+   ├── product.txt
+   └── ui
+       └── deployment.yaml
+
+
+Product information
+-------------------
+
+General product information about the packaged Solution must be stored in the
+``product.txt`` file, stored at the archive root.
+
+It must respect the following format::
+
+   NAME=${YOUR_SOLUTION_NAME}
+   VERSION=${YOUR_PROJECT_FULL_VERSION}
+
```

Comment:
I agree that requesting external services / Solutions from a Solution has valid use-cases, and we may want to invest time looking at possible ways to manage this. I believe it could become a part of some "Operator of Operators" that we'd like to add in the future, for instance something like OLM from CoreOS.

I'll add a note in these docs, but I don't have a clear answer right now.

However, regarding common services, such as Kafka+Zookeeper or ELK, I believe those should be part of the original offering from MetalK8s. Thoughts @NicolasT?

Reply:
Dependency handling is a big subject so it's ok as just a note for now.

If there are services to be provided by the platform, that the Solution expects (which ELK and Prometheus already are), maybe a Solution could mention it in its own manifest, as a capability name (for optional services, should that happen) or a minimum MetalK8s version (for services that will always be provided once introduced to the platform).

Reply:
I added a `REQUIRE_METALK8S` variable in the expected `product.txt`, which will allow for a minimal dependency management. Also added a note for handling more specific dependencies, such as monitoring components, TBD later.